src/dbnode/storage/index.go

```diff
@@ -1565,6 +1565,15 @@ func (i *nsIndex) ensureBlockPresentWithRLock(blockStart time.Time) (index.Block
 		return nil, i.unableToAllocBlockInvariantError(err)
 	}
 
+	// NB(bodu): Use same time barrier as `Tick` to make sealing of cold index blocks consistent.
+	// We need to seal cold blocks write away for cold writes.
+	lastSealableBlockStart := retention.FlushTimeEndForBlockSize(i.blockSize, i.nowFn().Add(-i.bufferPast))
```

Comment:
nit: Perhaps put this into a method on the index itself so guaranteed to be the same going forward?
```go
func (i *nsIndex) lastSealableBlockStart() time.Time {
  return retention.FlushTimeEndForBlockSize(i.blockSize, i.nowFn().Add(-i.bufferPast))
}
```